### PR TITLE
fix: only do column optimization on one partition of input dataset

### LINKED ISSUE #233: dask_awkward column optimization scales linearly with number of partitions in dataset
This causes major problems for the typically heavy HEP task graphs (for a complete dataset optimization times would be measured in hours).

The fix is something along the lines of only running the column optimization once per AwkwardInputLayer by artificially reducing the number of partition to analyze to 1 for the column optimization.

### PR DESCRIPTION
**Pull request by lgray**
Fixes #233

Posting a PR with @martindurant's patch to lodge here results of conversation / hacking in slack.

With some of the suggested changes in https://github.com/dask-contrib/dask-histogram/pull/74, it makes the complete analysis chain optimizable looking at only one partition per input dataset.

### CODE REVIEW

File: src/dask_awkward/#__init__.py# [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from dask_awkward import config  # isort:skip; load awkward config
```
Comment (martindurant) [inline src/dask_awkward/#__init__.py#:1]: This file is a debug remnant?

**Comment by lgray** [comment]
@martindurant what further testing needs to be done for this one (I've run it through the large analysis and everything works fine).

The only odd bit is `visualize()` run before `compute()` with this patch appears to somehow spoil the necessary columns (if I avoid the visualize or do `visualize(optimize_graph=True)` the problem disappears. Intensely weird but not critical IMO.

**Comment by lgray** [comment]
@martindurant what else is needed here?

File: tests/test_core.py [hunk 2]
```diff
@@ -324,20 +324,20 @@ def test_scalar_dtype() -> None:
...
-    s_dumped = pickle.dumps(c1)
-    c2 = pickle.loads(s_dumped)
-    assert_eq(c1, c2)
-    s1 = dak.sum(daa["points"]["y"], axis=None)
-    s_dumped = pickle.dumps(s1)
-    s2 = pickle.loads(s_dumped)
-    assert_eq(s1, s2)
-
-    assert s1.known_value is None
+# def test_scalar_pickle(daa: Array) -> None:
```
Comment (martindurant) [inline tests/test_core.py:327]: Why does this fail? Can it be reinstated? If not, we should either comment or remove it. I think I had a snippet to re-add the _meta objects to the layers, if that's what we need.

Comment (lgray) [reply tests/test_core.py:327]: Yes that snippet will resurrect that test

Comment (martindurant) [reply tests/test_core.py:327]: Do you need me to apply it?

Comment (lgray) [reply tests/test_core.py:327]: I am presently in a car so it would be appreciated :-)

**Pull request merged**